packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/PollingMetadata.java

```diff
@@ -0,0 +1,270 @@
+     * {@link ClientMethodsReturnDescription}.
+     *
+     * @return the polling metadata.
+     */
+    static PollingMetadata create(Operation operation, ProxyMethod proxyMethod, IType operationResponseType) {
+        final JavaSettings settings = JavaSettings.getInstance();
+        final PollingSettings pollingSettings = settings.getPollingSettings(proxyMethod.getOperationId());
+
+        if (pollingSettings == null) {
+            return null;
+        }
+
+        final HttpMethod httpMethod = MethodUtil.getHttpMethod(operation);
+        final LongRunningMetadata lroMetadata = operation.getLroMetadata();
+        if (lroMetadata == null || operation.getConvenienceApi() == null) {
+            // Create 'PollingMetadata' from 'PollingSettings'.
+            //
+            if (settings.isFluent()) {
+                return new PollingMetadata(pollingSettings.getPollingStrategy(),
+                    pollingSettings.getSyncPollingStrategy(), operationResponseType.asNullable(),
+                    operationResponseType.asNullable(), pollingSettings.getPollIntervalInSeconds(), httpMethod);
+            }
+
+            final IType pollResultType = getPollResultType(pollingSettings, operationResponseType);
+            final IType finalResultType;
+            if (httpMethod == HttpMethod.DELETE) {
+                finalResultType = PrimitiveType.VOID;
+            } else {
+                finalResultType = getFinalResultType(pollingSettings, operationResponseType);
+            }
+            return new PollingMetadata(pollingSettings.getPollingStrategy(), pollingSettings.getSyncPollingStrategy(),
+                pollResultType, finalResultType, pollingSettings.getPollIntervalInSeconds(), httpMethod);
+        } else {
+            // Create 'PollingMetadata' from TypeSpec long-running metadata.
+            // Note: Only TypeSpec would have 'Operation::LongRunningMetadata' available.
+            //
+            return create(lroMetadata, pollingSettings, httpMethod);
+        }
+    }
+
+    /**
+     * Gets the view of the polling metadata as {@link MethodPollingDetails}.
+     * <p>
+     * The resulting {@link MethodPollingDetails} is used for long-running {@link ClientMethod} i.e, client methods of
+     * type {@link ClientMethodType#LongRunningBeginSync}, {@link ClientMethodType#LongRunningBeginAsync},
+     * {@link ClientMethodType#LongRunningSync} and {@link ClientMethodType#LongRunningAsync}.
+     * </p>
+     *
+     * @return the polling metadata.
+     */
+    MethodPollingDetails asMethodPollingDetails() {
+        return new MethodPollingDetails(pollingStrategy, syncPollingStrategy, pollResultType, finalResultType,
+            pollIntervalInSeconds);
+    }
+
+    boolean hasBinaryDataResultTypes() {
+        if (ClassType.BINARY_DATA.equals(pollResultType)) {
+            return ClassType.BINARY_DATA.equals(finalResultType) || ClassType.VOID.equals(finalResultType.asNullable());
```

Comment:
Seems weird that `finalResultType == ClassType.VOID` indicates BinaryData result type.. Where is this method used?

Reply:
> Seems weird that `finalResultType == ClassType.VOID` indicates BinaryData result type.. Where is this method used?

Just realized, I'll remove this `hasBinaryDataResultTypes()` method since we're using `hasModelResultTypes()`.